modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportStartReindexJobAction.java

```diff
@@ -141,6 +168,56 @@ protected ClusterBlockException checkBlock(StartReindexJobAction.Request request
         return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
     }
 
+    private void createReindexTaskDoc(String taskId, ReindexRequest reindexRequest, boolean indexExists, ActionListener<Void> listener) {
+        if (indexExists) {
+            IndexRequest indexRequest = new IndexRequest(ReindexTask.REINDEX_INDEX).id(taskId);
+            try (XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON)) {
+                ReindexTaskIndexState reindexState = new ReindexTaskIndexState(reindexRequest);
+                reindexState.toXContent(builder, ToXContent.EMPTY_PARAMS);
+                indexRequest.source(builder);
+            } catch (IOException e) {
+                listener.onFailure(new ElasticsearchException("Couldn't serialize reindex request into XContent", e));
+            }
+            taskClient.index(indexRequest, new ActionListener<>() {
+                @Override
+                public void onResponse(IndexResponse indexResponse) {
+                    listener.onResponse(null);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    listener.onFailure(e);
+                }
+            });
+        } else {
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest();
+            createIndexRequest.settings(reindexIndexSettings());
+            createIndexRequest.index(ReindexTask.REINDEX_INDEX);
+            createIndexRequest.cause("auto(reindex api)");
```

Comment:
we should not have any analyzed fields on this index (at least for now), as we do not do any searches, just lookup by ID, so perhaps set `"dynamic": false` for the mapping.